lib/hidapi/udev.py

```diff
@@ -176,6 +176,32 @@ def find_paired_node(receiver_path, index, timeout):
     return None
 
 
+def find_paired_node_wpid(receiver_path, index):
+    """Find the node of a device paired with a receiver, get wpid from udev"""
```

Comment:
When I look at the output of `udevadm info -q -all /dev/hidrawN` I see the parent device
```
  looking at parent device '/devices/pci0000:00/0000:00:14.0/usb3/3-7/3-7:1.2/0003:046D:C52B.0007/0003:046D:4051.0008':
    KERNELS=="0003:046D:4051.0008"
    SUBSYSTEMS=="hid"
    DRIVERS=="logitech-hidpp-device"
    ATTRS{country}=="00"
```
It seems to me that getting this node and picking apart the KERNELS attribute would be easier.

Reply:
I try to use this approach, but can not find how to get KERNELS from udev, so I implement parse of device path instead. If you can do more pretty implementation - you are welcome :)

Reply:
But, as you say, there doesn't appear to be any way to get this field.

Reply:
I'm sure should be a way to access KERNELS field. At least udevadm does this.  Just I don't find how to do this.

Reply:
I think that this may be something that is missing from python udev.  I looked through what I think is the code and wasn't able to find anything that looked for this field.  Strange.

Reply:
But it might be a bit cleaner to get information from the hid device, which is already used to find the physical path.  One possibility is to use its device_path, which doesn't have the /dev/hidrawN at the end.  Another possibility is to use its HID_ID property, which is a bit strangely formatted but has the WPID in it.  Use ....properties.get('HID_ID')

Reply:
Thank you, I rewrite function ti use HID_ID and it now much simpler